File: Java_Generated/Java/untitled/src/measurements/Measurements.java
package measurements;

import category_theory.CategoryObject;
import category_theory.interfaces.IDesktop;
import measurements.interfaces.IMeasurement;
import measurements.interfaces.IMeasurements;

import java.util.ArrayList;
import java.util.List;

public class Measurements extends CategoryObject implements IMeasurements {
    public Measurements(String name, IDesktop desktop) {
        super(name, desktop);
    }

    List<IMeasurement> measurements = new ArrayList<>();

    @Override
    public int getMeasurementsCount() {
        return measurements.size();
    }

    @Override
    public IMeasurement getMeasurement(int i) {
        return measurements.get(i);
    }

    @Override
    public void updateMeasurements() {

    }

    @Override
    public void addMeasurement(IMeasurement measurement) {
        measurements.add(measurement);
    }
}

File: src/Python/measurements/measurements.py
from typing import List
from measurement import Measurement
from ..desktop import Desktop
from ..category_object import CategoryObject

class Measurements(CategoryObject):
    def __init__(self, name, desktop):
        super().__init__(name, desktop)
        self._measurements: List[Measurement] = []

    def __len__(self):
        return len(self._measurements)
    
    def __getitem__(self, i: int):
        return self._measurements[i]
    
    def update(self):
        pass

    def append(self, measurement: Measurement):
        self._measurements.append(measurement)


    def __add__(self, measurement: Measurement):
        self.append(measurement)
